backend/parsing/TestServer.java

```diff
@@ -0,0 +1,41 @@
+package parsing;
+
+import org.json.JSONObject;
+
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+// Just a very basic server to try out RPC requests and responses.
+public class TestServer {
+
+    public static void main(String[] args) throws IOException {
+
+        try (ServerSocket serverSocket = new ServerSocket(8080)) {
+
+            while (true) {
+                try (Socket clientSocket = serverSocket.accept();
+                     PrintWriter out = new PrintWriter(clientSocket.getOutputStream(), true);
+                     BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()))) {
+
+                    String requestString = in.readLine();
+
+                    JSONObject request = new JSONObject(requestString);
+
+                    JSONObject result = new JSONObject();
+                    result.put("grade", 20);
+                    result.put("message", "Hello world");
+
+                    JSONObject response = new JSONObject();
+                    response.put("jsonrpc", "2.0");
+                    response.put("id", request.get("id"));
+                    response.put("result", result);
+
+                    String responseString = response.toString();
+
+                    out.println(responseString);
```

Comment:
Should you verify the API key is correct, for illustration purposes, in the TestServer?

Reply:
You're right, I forgot to update the TestServer. Given we'll be changing most of this anyway, should I just leave it as it is for now?

Reply:
Sure, we can do this when we get back onto the API stuff.